app/components/Configuration.js

```diff
@@ -38,13 +40,11 @@ export default class Configuration extends Component<Props> {
     (this: any).goToNext = this.goToNext.bind(this);
   }
 
+  // configuration from disk is not yet available during componentDidMount
   componentDidMount() {
-    const { configuration, hideConfiguration } = this.props;
     // $FlowFixMe - flow thinks document.body could be undefined
     document.body.addEventListener('keydown', this.handleKeyDown);
-    if (configuration.policies.autostart) {
-      hideConfiguration();
-    }
+    isLaunching = true;
```

Comment:
This now works, but it's not guaranteed to work.

The calls as I see them are: `render -> componentDidMount -> render` and by luck, the `wakeUp` call, where configuration from disk is read, just happens before the second render... 

That's the reason why we cannot set `isLaunching = true` at the beginning of the file. That's why it could break at in the future.

Two options to make this deterministic:
- set `isLaunching=false` after one successful `autostart` did happen. (would have to rename the isLaunching, and also could be some weird behavior, where autostart is triggered once when you play with the setting)
- add a `wakeUpDone` flag into the configuration or somewhere else in the store/state, and only evaluate the autostart flag when wake up is finished.
- or somehow have the `hideConfiguration depending on autostart` call after wakeup, maybe outside of the configuration.js

Reply:
Let's leave this dirty solution for now and solve this for good once it's a HOC.